pupil_src/shared_modules/blink_detection.py

```diff
@@ -112,11 +112,16 @@ def recent_events(self, events={}):
         self.history.extend(events.get("pupil", []))
 
         try:  # use newest gaze point to determine age threshold
+            assert self.history[-1]["timestamp"] > self.history[0]["timestamp"]
```

Comment:
Do you need this assert once or should it rather be part of the while loop? Otherwise, I would go with something like
```
inconsistent_history = self.history[-1]["timestamp"] > self.history[0]["timestamp"]
if inconsistent_history:
    self.reset_history()
    return
```
I think asserts make most sense for "debugging" (i.e. you expect it to be false all the time), not for something that you expect can actually happen. The above code makes it imo clearer and is also faster (assert also always evaluates the condition). As an additional bonus, you no longer need the comment.

Reply:
Apparently it's also possible to just disable assert globally.

Reply:
You are right that we could keep the newest one.

Reply:
Unfortunately, I forgot that history is a `deque` does not allow sliced deletions. Which leaves us with two options:
1. `history.clear()`, single fast call, clears complete history
2. `history.popLeft()` until only one element remains

I favor option 1. heavily since
a) such big timebase changes rarely happen,
b) the history is comparably short and fills quite quickly,
c) option 2. would mean much more code

Reply:
agreed